lib/pavilion/test_run.py

```diff
@@ -1309,3 +1311,82 @@ def create_id_dir(id_dir):
 
     def __repr__(self):
         return "TestRun({s.name}-{s.id})".format(s=self)
+
+    def get_skipped(self):
+        match_list = []
+        match_list = self.get_match_only_if(match_list)
+        match_list = self.get_match_not_if(match_list)
+
+        try:
+            if len(match_list) is 0:
+                return False
+            else:
+                self.status.set(STATES.SKIPPED, match_list[0])
+                return True
+
+        except TypeError:
+            self.logger.error("Length of conditional match list not int")
+
+    def get_match_not_if(self, match_list):
+        """Get match not-if is the conditional check for not if statements
+         in the yaml config. It checks a variable for a match and will
+         return False if found.
+        :param match_list: A list of conditions from not-if resulting
+        in a SKIP."""
+
+        try:
+            variable_base = self.var_man
+            not_if_dict = self.config['not_if']
+        except KeyError:
+            self.logger.error("Error reading \'only_if\' key in config.")
+            return KeyError
+
+        try:
+            for key in not_if_dict:
+                real_key = variable_base[key]
+                for value in not_if_dict[key]:
+                    # If a key has a match log it.
+                    if real_key == value:
+                        message = ("Not if {0} is {1}. "
+                                   "The current {0} is {2}: SKIPPED"
+                                   .format(key, value, real_key))
+
+                        match_list.append(message)
+        except UnboundLocalError:
+            self.logger.error("Not_if_dict not assigned, check not_if key")
+
+        return match_list  # Return the list of conditional errors, can be None.
+
+    def get_match_only_if(self, match_list):
```

Comment:
You can merge this with the 'not_if' method. I'll come down in a bit and we can go over how.